From: Josef Bacik <jbacik@fusionio.com>
Date: Fri, 18 Oct 2013 12:10:36 -0400
Patch-mainline: 3.12
Git-commit: 1bda19eb73d68b304148e67253e47cef049a419d
Subject: [PATCH] Btrfs: release path before starting transaction in
 can_nocow_extent

We can't be holding tree locks while we try to start a transaction, we will
deadlock.  Thanks,

Reported-by: Sage Weil <sage@inktank.com>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: Chris Mason <chris.mason@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/inode.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -6469,6 +6469,7 @@ noinline int can_nocow_extent(struct btr
 
 	if (btrfs_extent_readonly(root, disk_bytenr))
 		goto out;
+	btrfs_release_path(path);
 
 	/*
 	 * look for other files referencing this extent, if we
